components/infobox/wikis/stormgate/infobox_item_custom.lua

```diff
@@ -0,0 +1,198 @@
+---
+-- @Liquipedia
+-- wiki=stormgate
+-- page=Module:Infobox/Item/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local Faction = require('Module:Faction')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Page = require('Module:Page')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local MessageBox = require('Module:Message box')
+
+local Injector = Lua.import('Module:Widget/Injector')
+local Item = Lua.import('Module:Infobox/Item')
+
+local Widgets = require('Module:Widget/All')
+local Cell = Widgets.Cell
+local Center = Widgets.Center
+local Title = Widgets.Title
+
+---@class StormgateItemInfobox: ItemInfobox
+---@field data table
+local CustomItem = Class.new(Item)
+local CustomInjector = Class.new(Injector)
+
+local ICON_DEPRECATED = '[[File:Cancelled Tournament.png|link=]]'
+local VALID_ITEMS = {
+	'Gear',
+}
+
+---@param frame Frame
+---@return Html
+function CustomItem.run(frame)
+	local item = CustomItem(frame)
+
+	assert(Table.includes(VALID_ITEMS, item.args.informationType), 'Missing or invalid "informationType"')
+
+	item:setWidgetInjector(CustomInjector(item))
+
+	self.data = {
+		introduced = self:_processPatchFromId(args.introduced),
+		deprecated = self:_processPatchFromId(args.deprecated),
+	}
+
+	local builtInfobox = item:createInfobox()
+
+	return mw.html.create()
+		:node(builtInfobox)
+		:node(CustomItem._deprecatedWarning(self.data.deprecated.display))
```

Comment:
for the fix you asked for:
```suggestion
	item.data = {
		introduced = item:_processPatchFromId(args.introduced),
		deprecated = item:_processPatchFromId(args.deprecated),
	}

	local builtInfobox = item:createInfobox()

	return mw.html.create()
		:node(builtInfobox)
		:node(CustomItem._deprecatedWarning(item.data.deprecated.display))
```
i.e. use `item` instead of `self` inside this function